### PR TITLE
ci: label pull requests based on their target branch

### PR DESCRIPTION
**Pull request by Lehoczky**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I've seen Daniel adding the `2.x` and `3.x` labels manually to PRs and wanted to help him out by labelling my own PR. Unfortunately, contributors don't have permission to edit PR labels, so I couldn't do that. 

This PR makes this process automated.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Lehoczky** [comment]
Looks like the `GITHUB_TOKEN` is missing some permissions. I'm not sure what is needed to add a label to a PR, but `pull_request: write` should be enough intuitively.

**Comment by danielroe** [comment]
It might be that the token on your fork doesn't have permissions to add labels on the main branch, or otherwise anyone could end up doing nefarious things.

I would suggest instead using the `pull_request_target` event which can run when the pull request is first opened, e.g.:

https://github.com/nuxt/nuxt/blob/9ef94d6afb83a067f16476290ecc1ad5e7070290/.github/workflows/semantic-pull-requests.yml#L3-L8

This will also run 'from' the main branch so shouldn't have any permission issues.

**Comment by Lehoczky** [comment]
Oh silly me, I haven't realized the token being used is from my fork 😅

I've updated the trigger as you suggested, thank you for the feedback!^^

**Review by danielroe — APPROVED** [review]

File: .github/workflows/label-pr.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  add-pr-label:
+    name: Add PR label
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.repository == 'nuxt/nuxt'
+    steps:
+      - uses: actions/github-script@v6
```
Comment (danielroe) [inline .github/workflows/label-pr.yml:19]:
```suggestion
      - uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**